### PR TITLE
UILabel Attribute

### PR DESCRIPTION
**Pull request by SwiftFoxx**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [ ] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.
- [x] I have added 2 new functions in `UILabelExtensions.swift`

### CODE REVIEW

File: Sources/SwifterSwift/UIKit/UILabelExtensions.swift [hunk 1]
```diff
@@ -34,6 +34,40 @@ public extension UILabel {
         label.sizeToFit()
         return label.frame.height
     }
+    
+    
+    /// SwifterSwift: Add attribute to UILabel
+    /// - Parameters:
+    ///   - range: The range of the string to apply attributes
+    func attribute(textIn range: Range<String.Index>?, _ attributes: [NSAttributedString.Key : Any]) {
+        guard let range = range else {
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UILabelExtensions.swift:43]: If these values can't be `nil` then why make them optional?

File: Sources/SwifterSwift/UIKit/UILabelExtensions.swift [hunk 2]
```diff
@@ -34,6 +34,40 @@ public extension UILabel {
         label.sizeToFit()
         return label.frame.height
     }
+    
+    
+    /// SwifterSwift: Add attribute to UILabel
+    /// - Parameters:
+    ///   - range: The range of the string to apply attributes
+    func attribute(textIn range: Range<String.Index>?, _ attributes: [NSAttributedString.Key : Any]) {
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UILabelExtensions.swift:42]: I think a better interface would be something like `setAttributes(_ attributes: [NSAttributedString.Key : Any], inRange: Range<String.Index>)`

File: Sources/SwifterSwift/UIKit/UILabelExtensions.swift [hunk 3]
```diff
@@ -34,6 +34,40 @@ public extension UILabel {
...
+        }
+        let attributedString = NSMutableAttributedString(string: text)
+        attributedString.addAttributes(attributes, range: NSRange(range, in: text))
+        attributedText = attributedString
+    }
+    
+    /// SwifterSwift: Add attribute to UILabel
+    /// - Parameters:
+    ///   - ranges: The ranges of the strings to apply attributes
+    func attribute(textIn ranges: Array<Range<String.Index>?>, _ attributes: [NSAttributedString.Key : Any]) {
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UILabelExtensions.swift:59]: Similar comments to above.

**Review by guykogus — CHANGES_REQUESTED** [review]
Thanks for the PR @SwiftFoxx. Left a few small comments and you also need to finish the checklist (especially CHANGELOG and tests).

File: Sources/SwifterSwift/UIKit/UILabelExtensions.swift (hunk 1 above)
Comment (SwiftFoxx) [reply Sources/SwifterSwift/UIKit/UILabelExtensions.swift:43]: Made those optional so the developer does't have to unwrap or check with conditions. 
Remove optional? Sure, I'll do that. Let me know.

File: Sources/SwifterSwift/UIKit/UILabelExtensions.swift (hunk 2 above)
Comment (SwiftFoxx) [reply Sources/SwifterSwift/UIKit/UILabelExtensions.swift:42]: Right, okay. Thanks!

File: Sources/SwifterSwift/UIKit/UILabelExtensions.swift (hunk 3 above)
Comment (SwiftFoxx) [reply Sources/SwifterSwift/UIKit/UILabelExtensions.swift:59]: Alright.

**Comment by SwiftFoxx** [comment]
> Thanks for the PR @SwiftFoxx. Left a few small comments and you also need to finish the checklist (especially CHANGELOG and tests).

Thanks for your review. I have made changes and pushed as requested. And also updated the checklist.

File: Sources/SwifterSwift/UIKit/UILabelExtensions.swift [hunk 4]
```diff
@@ -34,6 +34,35 @@ public extension UILabel {
...
+        }
+        let attributedString = NSMutableAttributedString(string: text)
+        attributedString.addAttributes(attributes, range: NSRange(range, in: text))
+        attributedText = attributedString
+    }
+    
+    /// SwifterSwift: Add attribute to UILabel
+    /// - Parameters:
+    ///   - ranges: The ranges of the strings to apply attributes
+    func addAttributes(_ attributes: [NSAttributedString.Key : Any], in ranges: Array<Range<String.Index>>) {
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UILabelExtensions.swift:55]:
```suggestion
    func addAttributes(_ attributes: [NSAttributedString.Key : Any], in ranges: [Range<String.Index>]) {
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -344,6 +344,8 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Added `createTemporaryDirectory()` to create a directory for saving temporary files. [#615](https://github.com/SwifterSwift/SwifterSwift/pull/615) by [guykogus](https://github.com/guykogus)
 - **UILabel**:
   - Added `init(text:style)` to create a `UILabel` with a text and font style. [#607](https://github.com/SwifterSwift/SwifterSwift/pull/607) by [marcocapano](https://github.com/marcocapano)
+  - Added `addAttribute(_:, in:)` to add attributed text easily.
+    [#1021](https://github.com/SwifterSwift/SwifterSwift/pull/1021) by [SwiftFoxx](https://github.com/SwiftFoxx)
```
Comment (guykogus) [inline CHANGELOG.md:348]:
```suggestion
  - Added `addAttributes(_:, in:)` to add attributes to text easily. [#1021](https://github.com/SwifterSwift/SwifterSwift/pull/1021) by [SwiftFoxx](https://github.com/SwiftFoxx)
```

**Review by guykogus — CHANGES_REQUESTED** [review]
Sorry it's been so long without a review.
Can you please add a test for covering an array of ranges?
And your tests don't actually do anything, you're not performing any assertions...

**Comment by viktart** [comment]
I think we don't need the whole new methods only for updating `UILabel.attributedText` property. I suggest we close this PR.

**Comment by guykogus** [comment]
Closing as this PR isn't being addressed properly by the author. Feel free to reopen if you ever want to continue with it.

**Pull request closed without merging**